### PR TITLE
Create a POC of an app to simplify deployment of Kubeflow

### PR DESCRIPTION
**Pull request by jlewi**
* The app is intended to optimize a Kubeflow app based on a user's
actual K8s setup.

* To illustrate this the app checks whether the user's cluster
has a default storage-class and uses that to determine whether to
enable PD for Jupyter notebooks.

* The intent is to publish this as a docker container which would
  also contain ksonnet.

* As described in the README a user should be able to start a shell
  in this container and walk through the steps of creating an App
  without having to install additional tools.

* This is currently using ksonnet at HEAD because they recently
  refactored the code and we want to take advantage of that; in particular
  they started organizing the code into actions.

* The code and generated app currently require a version of ks > 0.9.2; which means we currently need
 to build ks from source. We include a version of ks in the Docker image that is built from the appropriate
 commit.

Related to #105

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/595)

### CODE REVIEW

**Comment by kkasravi** [comment]
@jlewi i may of hit a similar issue - I had to checkout ksonnet-lib under the kubeflow app. 

I've checked out your PR and am running it - I'll update here.

**Comment by inc0** [comment]
Have you guys tried helm? To install helm you pull golang app "helm" and run helm init. That's it. Pretty cool ux

**Comment by kunmingg** [comment]
@jlewi 
For the "bootstrap/vendor", if it won't be updated very often (not much version control needed), can we have it stored in separate repo or cloud storage accessible by endpoint?

**Comment by jlewi** [comment]
@inc0 I've tried helm. I'm open to discussing helm but lets not do it in the context of this PR. I think to some extent the intent here is to do things that can't be solved by helm/ks. Take a look at my most recent comment on #105 to get an idea of where I want to go with it.

@kkasravi What issue are you refering to? ksonnet pushed a fix to ksonnet/ksonnet#420. I include a version of ks built at the appropriate commit in the docker image now.

@kunmingg checking in vendor is common practice for go.See kubeflow/tf-operator#222

File: bootstrap/developer_guide.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Developer guide for bootstrap
+
+## Building the Operator
+
+Create a symbolic link inside your GOPATH to the location you checked out the code
+
+```sh
+mkdir -p ${GOPATH}/src/github.com/kubeflow
+ln -sf ${GIT_KUBEFLOW} ${GOPATH}/src/github.com/kubeflow/kubeflow kubeflow
```
Comment (jose5918) [inline bootstrap/developer_guide.md]: extra arg here

File: bootstrap/developer_guide.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Install dependencies, `-v` will ignore subpackage vendor
+
+```sh
+glide install -v
+```
+
+Build it
+
+```sh
+go install github.com/kubeflow/tf-operator/cmd/tf-operator
```
Comment (jose5918) [inline bootstrap/developer_guide.md]: go install github.com/kubeflow/kubeflow/bootstrap/cmd/bootstrap?

File: bootstrap/developer_guide.md (hunk 1 above)
Comment (jlewi) [reply bootstrap/developer_guide.md]: Fixed

File: bootstrap/developer_guide.md (hunk 2 above)
Comment (jlewi) [reply bootstrap/developer_guide.md]: Fixed

**Comment by kunmingg** [comment]
@jlewi 
Thanks, and one more question:
```-v ${HOME}:/home/${USER}```
It might be aggressive to access ${HOME},
How about narrow it down to only sub-dirs it need to access?

**Comment by jlewi** [comment]
@kunmingg Why is it aggressive to map in home directory? This is no different than user running a binary directly. The reason for mapping in the home directory is to

- Let the user store the app in their home directory
- Make use of config files for .kube and .gcloud

**Comment by jlewi** [comment]
@inc0 @kunmingg Anymore comments? If not would you mind approving? Would be good to get this submitted so we can follow up with more improvements.

**Comment by jlewi** [comment]
@kunmingg you need to "/lgtm"; only approvers can "/approve"

**Pull request merged**